core/src/main/scala/cats/instances/list.scala

```diff
@@ -82,6 +82,10 @@ trait ListInstances extends cats.kernel.instances.ListInstances {
 
       override def foldM[G[_], A, B](fa: List[A], z: B)(f: (B, A) => G[B])(implicit G: Monad[G]): G[B] =
         Foldable.iteratorFoldM(fa.toIterator, z)(f)
+
+      override def fold[A](fa: List[A])(implicit A: Monoid[A]): A = A.combineAll(fa)
```

Comment:
I probably missed something but I am curious what gain we get from overriding this? I mean the 
In `Monoid.combineAll` it's 
```scala 
as.foldLeft(empty)(combine)
```
In foldable.fold it's
```Scala
 foldLeft(fa, A.empty) { (acc, a) =>
      A.combine(acc, a)
    }
```